<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2021-08-24" />

<title>SampleQC: Generating plots for paper</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SampleQC_proj</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">SampleQC: Generating plots for paper</h1>
<h4 class="author">Will Macnair</h4>
<address class="author_afil">
Institute for Molecular Life Sciences, University of Zurich, SwitzerlandSwiss Institute of Bioinformatics (SIB), University of Zurich, Switzerland<br><h4 class="date">August 24, 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>SampleQC_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongobjectspresent"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Environment:</strong> objects present </a>
</p>
</div>
<div id="strongEnvironmentstrongobjectspresent" class="panel-collapse collapse">
<div class="panel-body">
<p>
The global environment had objects present when the code in the R Markdown file was run. These objects can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment. Use <code>wflow_publish</code> or <code>wflow_build</code> to ensure that the code is always run in an empty environment.
</p>
<p>
The following objects were defined in the global environment when these results were created:
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
Name
</th>
<th style="text-align:left;">
Class
</th>
<th style="text-align:left;">
Size
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
q
</td>
<td style="text-align:left;">
function
</td>
<td style="text-align:left;">
1008 bytes
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210311code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210311)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210311code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210311)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomwmacnairSampleQCanalysistree0192172cb2a1bf18ec01ce2f3527b3c82a3da80btargetblank0192172a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/wmacnair/SampleQC_analysis/tree/0192172cb2a1bf18ec01ce2f3527b3c82a3da80b" target="_blank">0192172</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomwmacnairSampleQCanalysistree0192172cb2a1bf18ec01ce2f3527b3c82a3da80btargetblank0192172a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/wmacnair/SampleQC_analysis/tree/0192172cb2a1bf18ec01ce2f3527b3c82a3da80b" target="_blank">0192172</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rprofile
    Ignored:    analysis/qc01_prep_data_cache/
    Ignored:    analysis/qc02_distns_cache/
    Ignored:    analysis/qc03_sims_cache/
    Ignored:    data/miqc/
    Ignored:    output/qc07_comparisons/

Untracked files:
    Untracked:  _workflowr.yml
    Untracked:  analysis/.qc04_real.Rmd.swp
    Untracked:  analysis/.qc05_citeseq.Rmd.swp
    Untracked:  analysis/.qc06_ks.Rmd.swp
    Untracked:  analysis/.qc07_comparisons.Rmd.swp
    Untracked:  analysis/figure/
    Untracked:  output/qc03_sims/mahalanobis_qq_plot_sim.pdf
    Untracked:  output/qc03_sims/mahalanobis_qq_plot_sim.png

Unstaged changes:
    Modified:   analysis/qc01_prep_data.Rmd
    Modified:   analysis/qc02_distns.Rmd
    Modified:   analysis/qc03_sims.Rmd
    Modified:   analysis/qc04_real.Rmd
    Modified:   code/qc03_sims.R
    Modified:   output/qc01_prep_data/Campbell_qc_dt.txt.gz
    Modified:   output/qc01_prep_data/HGSOC_qc_dt.txt.gz
    Modified:   output/qc01_prep_data/Macosko_qc_dt.txt.gz
    Modified:   output/qc01_prep_data/Shekhar_qc_dt.txt.gz
    Modified:   output/qc01_prep_data/Wang_qc_dt.txt.gz
    Modified:   output/qc01_prep_data/Zeisel_qc_dt.txt.gz
    Modified:   output/qc02_distns/selected_biaxials.pdf
    Modified:   output/qc02_distns/selected_biaxials.png
    Modified:   output/qc02_distns/selected_marginals.pdf
    Modified:   output/qc02_distns/selected_marginals.png
    Modified:   output/qc03_sims/outs_scater_sim_ok_multimodal.pdf
    Modified:   output/qc03_sims/outs_scater_sim_ok_multimodal.png
    Modified:   output/qc03_sims/outs_scater_sim_ok_unimodal.pdf
    Modified:   output/qc03_sims/outs_scater_sim_ok_unimodal.png
    Modified:   output/qc03_sims/outs_scater_sim_out_multimodal.pdf
    Modified:   output/qc03_sims/outs_scater_sim_out_multimodal.png
    Modified:   output/qc03_sims/outs_scater_sim_out_unimodal.pdf
    Modified:   output/qc03_sims/outs_scater_sim_out_unimodal.png
    Modified:   output/qc03_sims/per_celltype_bias_abs.pdf
    Modified:   output/qc03_sims/per_celltype_bias_abs.png
    Modified:   output/qc03_sims/per_celltype_bias_prop.pdf
    Modified:   output/qc03_sims/per_celltype_bias_prop.png
    Modified:   output/qc03_sims/pr_curves_by_modality.pdf
    Modified:   output/qc03_sims/pr_curves_by_modality.png
    Modified:   output/qc03_sims/pr_curves_by_sample.pdf
    Modified:   output/qc03_sims/pr_curves_by_sample.png
    Modified:   output/qc03_sims/pr_curves_overall.pdf
    Modified:   output/qc03_sims/pr_curves_overall.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/qc04_real.Rmd</code>) and HTML (<code>docs/qc04_real.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wmacnair/SampleQC_analysis/blob/86834a17ef4cf47ae44e2be6a49b75fd445c5855/analysis/qc04_real.Rmd" target="_blank">86834a1</a>
</td>
<td>
wmacnair
</td>
<td>
2021-08-10
</td>
<td>
First commit of everything
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup-definitions" class="section level1">
<h1>Setup / definitions</h1>
<div id="libraries" class="section level2">
<h2>Libraries</h2>
<pre class="r"><code>library(&#39;SampleQC&#39;)

library(&#39;assertthat&#39;)
library(&#39;ggplot2&#39;)
library(&#39;patchwork&#39;)
library(&#39;scales&#39;)
library(&#39;stringr&#39;)
library(&#39;forcats&#39;)
library(&#39;data.table&#39;)
library(&#39;magrittr&#39;)
seed    = 20200725
set.seed(seed)</code></pre>
</div>
<div id="inputs" class="section level2">
<h2>Inputs</h2>
<pre class="r"><code># where is snRNAseq file?
qc_f        = &#39;data/snRNAseq/sn_qc_dt_anon_2021-07-25.txt.gz&#39;
qc_obj_f    = &#39;data/snRNAseq/sampleQC_obj_2021-07-25.rds&#39;

qc_names    = c(&#39;log_counts&#39;, &#39;logit_mito&#39;, &#39;splice_ratio&#39;)
annots_disc = c(&#39;donor_id&#39;, &#39;condition&#39;, &#39;batch&#39;, &#39;source&#39;)
annots_cont = c(&#39;med_splice&#39;)</code></pre>
</div>
<div id="helper-functions" class="section level2">
<h2>Helper functions</h2>
<pre class="r"><code>source(&#39;code/qc03_sims.R&#39;)</code></pre>
</div>
<div id="outputs" class="section level2">
<h2>Outputs</h2>
<pre class="r"><code># where to save?
save_dir  = &#39;output/qc04_real&#39;
if (!dir.exists(save_dir))
  dir.create(save_dir)

# annotations to save
disc_ls         = c(&#39;group_id&#39;, &#39;batch&#39;, &#39;mito_cat&#39;, &#39;N_cat&#39;)
cont_ls         = c(&#39;med_counts&#39;, &#39;med_splice&#39;)
annot_ls        = c(disc_ls, cont_ls)

# exclude clusters with above this splice_ratio
splice_cut      = 3

# what figs to save?
formats         = c(&#39;png&#39;, &#39;pdf&#39;)
outs_scater_f   = file.path(save_dir, &#39;outs_scater_bio.%s&#39;)
outs_distns_f   = file.path(save_dir, &#39;outs_distns_bio.%s&#39;)
kept_distns_f   = file.path(save_dir, &#39;kept_distns_bio.%s&#39;)
maha_qqs_f      = file.path(save_dir, &#39;mahalanobis_qq_plot_bio.%s&#39;)</code></pre>
</div>
</div>
<div id="load-inputs" class="section level1">
<h1>Load inputs</h1>
</div>
<div id="processing-calculations" class="section level1">
<h1>Processing / calculations</h1>
<pre class="r"><code>qc_dt       = qc_f %&gt;% fread %&gt;%
  .[, med_splice := median(splice_ratio), by=sample_id] %&gt;%
  make_qc_dt(qc_names = qc_names)</code></pre>
<pre class="r"><code># K_list    = c(4, 4, 3, 2, 1, 3, 4, 4, 4)
K_list    = c(3, 4, 1, 4, 2, 1, 3, 4)
# K_list    = c(1, 1, 1, 1, 1, 1, 1, 1)
if (file.exists(qc_obj_f)) {
  qc_obj    = readRDS(qc_obj_f)
} else {
  # find sample groups
  set.seed(seed)
  qc_obj    = calc_pairwise_mmds(qc_dt, qc_names, 
    annots_disc = annots_disc, annots_cont = annots_cont, 
    n_cores = 16)
  print(table(colData(qc_obj)$group_id))

  # fit model
  set.seed(seed)
  qc_obj    = fit_sampleqc(qc_obj, K_list = K_list, mcd_alpha = 0.3)

  # save outputs
  make_sampleqc_report(qc_obj, save_dir, &#39;snRNAseq&#39;)

  # specify outlier cluster
  saveRDS(qc_obj, file = qc_obj_f)
}</code></pre>
<pre class="r"><code># extract component means
group_ids   = levels(colData(qc_obj)$group_id)
beta_ks     = group_ids %&gt;%
  lapply(function(g) {
    beta_k  = metadata(qc_obj)$fit_list[[g]]$beta_k
    mu_0    = metadata(qc_obj)$fit_list[[g]]$mu_0
    beta_k  = sweep(beta_k, 2, mu_0, &#39;+&#39;)
    assert_that( all(abs(colMeans(beta_k) - mu_0) &lt; 1e-10) )
    return(beta_k)
  })

# get clusters with positive splice ratio
splice_idx    = qc_names == &#39;splice_ratio&#39;
exc_clusters  = lapply(beta_ks, 
  function(beta_k_mat) which(beta_k_mat[, splice_idx] &gt; splice_cut)
  ) %&gt;% setNames(group_ids)
exc_clusters  = exc_clusters[ sapply(exc_clusters, function(x) length(x) &gt; 0) ]

# get outliers
outliers_dt   = get_outliers(qc_obj, exc_clusters = exc_clusters)</code></pre>
<pre class="r"><code># calc scater outliers
scater_dt = calc_scater_dt(qc_dt)
set.seed(123)
miqc_dt   = calc_miqc_dt(qc_dt)</code></pre>
<pre><code>(only one component identified)</code></pre>
</div>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
<div id="embeddings-of-samples" class="section level2 tabset">
<h2 class="tabset">Embeddings of samples</h2>
<pre class="r"><code>for (sel_embed in c(&#39;UMAP&#39;, &#39;MDS&#39;)) {
  cat(&#39;### &#39;, sel_embed, &#39;\n&#39;)
  g_list  = lapply(seq_along(annot_ls), function(ii) {
    annot_i   = annot_ls[[ii]]
    var_type  = ifelse(annot_i %in% disc_ls, &#39;discrete&#39;, &#39;continuous&#39;)
    SampleQC:::.plot_one_embedding(qc_obj, annot_i, var_type, sel_embed) +
      labs( title=LETTERS[[ii]] )
    })
  g       = wrap_plots(g_list, ncol=2)
  print(g)
  png_f   = sprintf(&#39;%s/real_qc_embed_disc_%s.png&#39;, save_dir, sel_embed)
  ggsave(png_f, g, h=9, w=8)
  cat(&#39;\n\n&#39;)
}</code></pre>
<div id="umap" class="section level3">
<h3>UMAP</h3>
<p><img src="figure/qc04_real.Rmd/plot_embeddings_disc-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="mds" class="section level3">
<h3>MDS</h3>
<p><img src="figure/qc04_real.Rmd/plot_embeddings_disc-2.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="plot-sampleqc-model-fits-and-outliers-over-qc-biaxials" class="section level2">
<h2>Plot <code>SampleQC</code> model fits and outliers over QC biaxials</h2>
<pre class="r"><code>group_list  = levels(qc_obj$group_id)
for (g in group_list) {
  cat(&#39;### &#39;, g, &#39;{.tabset}\n&#39;)
  # which samples?
  samples_g   = sort(colData(qc_obj)$sample_id[ colData(qc_obj)$group_id == g ])
  # # just do a handful
  # samples_g   = sample(samples_g, min(5, length(samples_g))) %&gt;% sort
  for (s in samples_g) {
    cat(&#39;#### &#39;, s, &#39; \n&#39;)
    g_fit   = plot_fit_over_biaxials(qc_obj, s)
    g_out   = plot_outliers(qc_obj, s)
    g     = g_fit / g_out
    print(g)
    cat(&#39;\n\n&#39;)
  }
}</code></pre>
</div>
<div id="plot-outliers" class="section level2">
<h2>Plot outliers</h2>
<pre class="r"><code># assemble plot
s       = &#39;sample017&#39;
g_fit   = plot_fit_over_biaxials(qc_obj, s) + 
  labs( title = &#39;A. SampleQC fits&#39; )
g_maha  = plot_maha_dists(qc_obj, s) + 
  labs( title = &#39;B. Mahalanobis distances&#39; )
g_qc    = plot_outliers(qc_obj, s, outliers_dt) + 
  labs( title = &#39;C. SampleQC outliers&#39; )
g_scat  = plot_outliers(qc_obj, s, scater_dt) + 
  labs( title = &#39;D. scater outliers&#39; )
g_miqc  = plot_outliers(qc_obj, s, miqc_dt) + 
  labs( title = &#39;E. miQC outliers&#39; )
g       = list(g_fit, g_maha, plot_spacer(), g_qc, g_scat, g_miqc) %&gt;% 
  wrap_plots(ncol = 2, byrow = FALSE)

# save files
for (f in formats)
  ggsave(sprintf(outs_scater_f, f), g, h = 8, w = 10)

# plot
print(g)</code></pre>
<p><img src="figure/qc04_real.Rmd/plot_outliers-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-mahalanobis-distances" class="section level2">
<h2>Plot Mahalanobis distances</h2>
<pre class="r"><code>set.seed(seed)
sel_samples   = qc_obj$sample_id %&gt;% sample(6)
mahas_dt      = calc_mahas_dt(qc_obj)

# do plot
g             = plot_maha_qqs(mahas_dt, sel_samples)
for (f in formats)
  ggsave(sprintf(maha_qqs_f, f, f), g, h = 5, w = 8)
print(g)</code></pre>
<p><img src="figure/qc04_real.Rmd/plot_maha_qq-1.png" width="576" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/local/R/R-4.0.0/lib/libRblas.so
LAPACK: /usr/local/R/R-4.0.0/lib/libRlapack.so

locale:
 [1] LC_CTYPE=C                 LC_NUMERIC=C              
 [3] LC_TIME=en_CA.UTF-8        LC_COLLATE=en_CA.UTF-8    
 [5] LC_MONETARY=en_CA.UTF-8    LC_MESSAGES=en_CA.UTF-8   
 [7] LC_PAPER=en_CA.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_CA.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] flexmix_2.3-17              lattice_0.20-41            
 [3] scater_1.18.3               SingleCellExperiment_1.12.0
 [5] SummarizedExperiment_1.20.0 Biobase_2.50.0             
 [7] GenomicRanges_1.42.0        GenomeInfoDb_1.26.2        
 [9] IRanges_2.24.1              S4Vectors_0.28.1           
[11] BiocGenerics_0.36.0         MatrixGenerics_1.2.1       
[13] matrixStats_0.60.0          viridis_0.5.1              
[15] viridisLite_0.4.0           RColorBrewer_1.1-2         
[17] magrittr_2.0.1              data.table_1.14.0          
[19] forcats_0.5.1               stringr_1.4.0              
[21] scales_1.1.1                patchwork_1.1.1            
[23] ggplot2_3.3.5               assertthat_0.2.1           
[25] SampleQC_0.6.1              BiocStyle_2.18.1           
[27] colorout_1.2-2              BiocManager_1.30.10        
[29] workflowr_1.6.2            

loaded via a namespace (and not attached):
 [1] segmented_1.3-4           bitops_1.0-6             
 [3] fs_1.5.0                  rprojroot_2.0.2          
 [5] tools_4.0.0               irlba_2.3.3              
 [7] utf8_1.2.2                R6_2.5.0                 
 [9] vipor_0.4.5               uwot_0.1.10              
[11] DBI_1.1.1                 colorspace_2.0-2         
[13] nnet_7.3-14               withr_2.4.2              
[15] tidyselect_1.1.0          gridExtra_2.3            
[17] compiler_4.0.0            git2r_0.28.0             
[19] BiocNeighbors_1.8.2       DelayedArray_0.16.2      
[21] labeling_0.4.2            mvtnorm_1.1-2            
[23] mvnfast_0.2.7             digest_0.6.27            
[25] mixtools_1.2.0            R.utils_2.10.1           
[27] rmarkdown_2.10            XVector_0.30.0           
[29] pkgconfig_2.0.3           htmltools_0.5.1.1        
[31] sparseMatrixStats_1.2.0   highr_0.9                
[33] rlang_0.4.11              DelayedMatrixStats_1.12.2
[35] farver_2.1.0              generics_0.1.0           
[37] mclust_5.4.7              BiocParallel_1.24.1      
[39] gtools_3.9.2              R.oo_1.24.0              
[41] dplyr_1.0.3               BiocSingular_1.6.0       
[43] RCurl_1.98-1.2            modeltools_0.2-23        
[45] GenomeInfoDbData_1.2.4    scuttle_1.0.4            
[47] Matrix_1.3-2              ggbeeswarm_0.6.0         
[49] Rcpp_1.0.7                munsell_0.5.0            
[51] fansi_0.5.0               R.methodsS3_1.8.1        
[53] lifecycle_1.0.0           stringi_1.7.3            
[55] whisker_0.4               yaml_2.2.1               
[57] MASS_7.3-53               zlibbioc_1.36.0          
[59] grid_4.0.0                promises_1.1.1           
[61] crayon_1.4.1              beachmat_2.6.4           
[63] splines_4.0.0             knitr_1.33               
[65] pillar_1.6.2              igraph_1.2.6             
[67] codetools_0.2-18          glue_1.4.2               
[69] evaluate_0.14             vctrs_0.3.8              
[71] httpuv_1.5.5              gtable_0.3.0             
[73] purrr_0.3.4               kernlab_0.9-29           
[75] xfun_0.25                 rsvd_1.0.3               
[77] later_1.1.0.1             survival_3.2-7           
[79] tibble_3.1.3              beeswarm_0.2.3           
[81] ellipsis_0.3.2           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
