<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2021-08-24" />

<title>SampleQC: Analysis of CITE-seq data</title>

<script src="site_libs/header-attrs-2.10/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SampleQC_proj</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">SampleQC: Analysis of CITE-seq data</h1>
<h4 class="author">Will Macnair</h4>
<address class="author_afil">
Institute for Molecular Life Sciences, University of Zurich, SwitzerlandSwiss Institute of Bioinformatics (SIB), University of Zurich, Switzerland<br><h4 class="date">August 24, 2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-24
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>SampleQC_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongobjectspresent"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Environment:</strong> objects present </a>
</p>
</div>
<div id="strongEnvironmentstrongobjectspresent" class="panel-collapse collapse">
<div class="panel-body">
<p>
The global environment had objects present when the code in the R Markdown file was run. These objects can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment. Use <code>wflow_publish</code> or <code>wflow_build</code> to ensure that the code is always run in an empty environment.
</p>
<p>
The following objects were defined in the global environment when these results were created:
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
Name
</th>
<th style="text-align:left;">
Class
</th>
<th style="text-align:left;">
Size
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
q
</td>
<td style="text-align:left;">
function
</td>
<td style="text-align:left;">
1008 bytes
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210311code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210311)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210311code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210311)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomwmacnairSampleQCanalysistree0192172cb2a1bf18ec01ce2f3527b3c82a3da80btargetblank0192172a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/wmacnair/SampleQC_analysis/tree/0192172cb2a1bf18ec01ce2f3527b3c82a3da80b" target="_blank">0192172</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomwmacnairSampleQCanalysistree0192172cb2a1bf18ec01ce2f3527b3c82a3da80btargetblank0192172a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/wmacnair/SampleQC_analysis/tree/0192172cb2a1bf18ec01ce2f3527b3c82a3da80b" target="_blank">0192172</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rprofile
    Ignored:    analysis/qc01_prep_data_cache/
    Ignored:    analysis/qc02_distns_cache/
    Ignored:    analysis/qc03_sims_cache/
    Ignored:    analysis/qc04_real_cache/
    Ignored:    data/miqc/
    Ignored:    output/qc07_comparisons/

Untracked files:
    Untracked:  README.md
    Untracked:  _workflowr.yml
    Untracked:  analysis/.qc05_citeseq.Rmd.swp
    Untracked:  analysis/.qc06_ks.Rmd.swp
    Untracked:  analysis/.qc07_comparisons.Rmd.swp
    Untracked:  analysis/figure/
    Untracked:  code/.qc05_citeseq.R.swp
    Untracked:  output/qc03_sims/mahalanobis_qq_plot_sim.pdf
    Untracked:  output/qc03_sims/mahalanobis_qq_plot_sim.png

Unstaged changes:
    Modified:   analysis/qc01_prep_data.Rmd
    Modified:   analysis/qc02_distns.Rmd
    Modified:   analysis/qc03_sims.Rmd
    Modified:   analysis/qc04_real.Rmd
    Modified:   analysis/qc05_citeseq.Rmd
    Modified:   code/qc03_sims.R
    Modified:   code/qc05_citeseq.R
    Modified:   output/qc01_prep_data/Campbell_qc_dt.txt.gz
    Modified:   output/qc01_prep_data/HGSOC_qc_dt.txt.gz
    Modified:   output/qc01_prep_data/Macosko_qc_dt.txt.gz
    Modified:   output/qc01_prep_data/Shekhar_qc_dt.txt.gz
    Modified:   output/qc01_prep_data/Wang_qc_dt.txt.gz
    Modified:   output/qc01_prep_data/Zeisel_qc_dt.txt.gz
    Modified:   output/qc02_distns/selected_biaxials.pdf
    Modified:   output/qc02_distns/selected_biaxials.png
    Modified:   output/qc02_distns/selected_marginals.pdf
    Modified:   output/qc02_distns/selected_marginals.png
    Modified:   output/qc03_sims/outs_scater_sim_ok_multimodal.pdf
    Modified:   output/qc03_sims/outs_scater_sim_ok_multimodal.png
    Modified:   output/qc03_sims/outs_scater_sim_ok_unimodal.pdf
    Modified:   output/qc03_sims/outs_scater_sim_ok_unimodal.png
    Modified:   output/qc03_sims/outs_scater_sim_out_multimodal.pdf
    Modified:   output/qc03_sims/outs_scater_sim_out_multimodal.png
    Modified:   output/qc03_sims/outs_scater_sim_out_unimodal.pdf
    Modified:   output/qc03_sims/outs_scater_sim_out_unimodal.png
    Modified:   output/qc03_sims/per_celltype_bias_abs.pdf
    Modified:   output/qc03_sims/per_celltype_bias_abs.png
    Modified:   output/qc03_sims/per_celltype_bias_prop.pdf
    Modified:   output/qc03_sims/per_celltype_bias_prop.png
    Modified:   output/qc03_sims/pr_curves_by_modality.pdf
    Modified:   output/qc03_sims/pr_curves_by_modality.png
    Modified:   output/qc03_sims/pr_curves_by_sample.pdf
    Modified:   output/qc03_sims/pr_curves_by_sample.png
    Modified:   output/qc03_sims/pr_curves_overall.pdf
    Modified:   output/qc03_sims/pr_curves_overall.png
    Modified:   output/qc04_real/mahalanobis_qq_plot_bio.pdf
    Modified:   output/qc04_real/mahalanobis_qq_plot_bio.png
    Modified:   output/qc04_real/outs_scater_bio.pdf
    Modified:   output/qc04_real/outs_scater_bio.png
    Modified:   output/qc04_real/real_qc_embed_disc_MDS.png
    Modified:   output/qc04_real/real_qc_embed_disc_UMAP.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/qc05_citeseq.Rmd</code>) and HTML (<code>docs/qc05_citeseq.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wmacnair/SampleQC_analysis/blob/86834a17ef4cf47ae44e2be6a49b75fd445c5855/analysis/qc05_citeseq.Rmd" target="_blank">86834a1</a>
</td>
<td>
wmacnair
</td>
<td>
2021-08-10
</td>
<td>
First commit of everything
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup-definitions" class="section level1">
<h1>Setup / definitions</h1>
<div id="libraries" class="section level2">
<h2>Libraries</h2>
<pre class="r"><code>set.seed(20210104)</code></pre>
</div>
<div id="inputs" class="section level2">
<h2>Inputs</h2>
</div>
<div id="helper-functions" class="section level2">
<h2>Helper functions</h2>
<pre class="r"><code>library(&#39;SampleQC&#39;)
source(&#39;code/qc05_citeseq.R&#39;)</code></pre>
<pre><code>Loading SampleQC</code></pre>
<pre><code>Warning: S3 methods &#39;make_qc_dt.SingleCellExperiment&#39;, &#39;make_qc_dt.data.frame&#39;
were declared in NAMESPACE but not found</code></pre>
</div>
<div id="outputs" class="section level2">
<h2>Outputs</h2>
<pre class="r"><code># where to save?
save_dir  = &#39;output/qc05_citeseq&#39;
if (!dir.exists(save_dir))
  dir.create(save_dir)
qc_f          = file.path(save_dir, &#39;citeseq_qc_obj.rds&#39;)
cite_fits_f   = file.path(save_dir, &#39;cite_seq_qc_metric_grid.png&#39;)</code></pre>
</div>
</div>
<div id="load-inputs" class="section level1">
<h1>Load inputs</h1>
</div>
<div id="processing-calculations" class="section level1">
<h1>Processing / calculations</h1>
<pre class="r"><code>sce       = KotliarovPBMCData()</code></pre>
<pre><code>snapshotDate(): 2020-10-27</code></pre>
<pre><code>see ?scRNAseq and browseVignettes(&#39;scRNAseq&#39;) for documentation</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>see ?scRNAseq and browseVignettes(&#39;scRNAseq&#39;) for documentation</code></pre>
<pre><code>loading from cache</code></pre>
<pre><code>see ?scRNAseq and browseVignettes(&#39;scRNAseq&#39;) for documentation</code></pre>
<pre><code>loading from cache</code></pre>
<pre class="r"><code>qc_dt     = get_qc_from_cite(sce) %&gt;%
  setnames(
    c(&#39;log_counts_rna&#39;, &#39;log_feats_rna&#39;),
    c(&#39;log_counts&#39;, &#39;log_feats&#39;)
    ) %&gt;%
  make_qc_dt</code></pre>
<pre class="r"><code>obj_list  = list(
  negative  = neg_obj,
  lupus     = test_obj
  )
qc_dt     = names(obj_list) %&gt;%
  lapply(function(n) get_qc_from_cite(obj_list[[n]], n)) %&gt;%
  rbindlist %&gt;%
  .[, sample_id := paste0(experiment, &#39;_&#39;, tenx_lane) ]</code></pre>
<pre class="r"><code>qc_names    = c(&#39;log_counts&#39;, &#39;log_feats&#39;, &#39;logit_mito&#39;, &#39;log_counts_adt&#39;, &#39;log_feats_adt&#39;)

if (file.exists(qc_f)) {
  qc_obj    = readRDS(qc_f)
} else {
  # define QC metrics to use
  annots_disc = c(&#39;batch&#39;, &#39;tenx_lane&#39;)
  qc_obj      = calc_pairwise_mmds(qc_dt, qc_names, 
    annots_disc = annots_disc, annots_cont = NULL, n_cores = 4)
  print(table(colData(qc_obj)$group_id))

  qc_obj      = fit_sampleqc(qc_obj, K_all=2)
  saveRDS(qc_obj, file=qc_f)  
}</code></pre>
</div>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
<div id="plot-qc-metrics-by-group" class="section level2 tabset">
<h2 class="tabset">Plot QC metrics by group</h2>
<pre class="r"><code>group_list  = metadata(qc_obj)$group_list
for (g in group_list) {
  cat(&#39;## &#39;, g, &#39;{.tabset}\n&#39;)
  # which samples?
  samples_g   = sort(colData(qc_obj)$sample_id[ colData(qc_obj)$group_id == g ])
  for (s in samples_g) {
    cat(&#39;### &#39;, s, &#39; \n&#39;)
    g_fit       = plot_fit_over_biaxials(qc_obj, s)
    g_out       = plot_outliers(qc_obj, s)
    g           = g_fit / g_out
    print(g)
    cat(&#39;\n\n&#39;)
  }
}</code></pre>
</div>
<div id="sg1" class="section level2 tabset">
<h2 class="tabset">SG1</h2>
<div id="batch1_h1b1ln1" class="section level3">
<h3>batch1_H1B1ln1</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_histograms_split-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch1_h1b1ln2" class="section level3">
<h3>batch1_H1B1ln2</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_histograms_split-2.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch1_h1b1ln3" class="section level3">
<h3>batch1_H1B1ln3</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_histograms_split-3.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch1_h1b1ln4" class="section level3">
<h3>batch1_H1B1ln4</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_histograms_split-4.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch1_h1b1ln5" class="section level3">
<h3>batch1_H1B1ln5</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_histograms_split-5.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch1_h1b1ln6" class="section level3">
<h3>batch1_H1B1ln6</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_histograms_split-6.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="sg2" class="section level2 tabset">
<h2 class="tabset">SG2</h2>
<div id="batch2_h1b2ln1" class="section level3">
<h3>batch2_H1B2ln1</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_histograms_split-7.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch2_h1b2ln2" class="section level3">
<h3>batch2_H1B2ln2</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_histograms_split-8.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch2_h1b2ln3" class="section level3">
<h3>batch2_H1B2ln3</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_histograms_split-9.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch2_h1b2ln4" class="section level3">
<h3>batch2_H1B2ln4</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_histograms_split-10.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch2_h1b2ln5" class="section level3">
<h3>batch2_H1B2ln5</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_histograms_split-11.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch2_h1b2ln6" class="section level3">
<h3>batch2_H1B2ln6</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_histograms_split-12.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="plot-pairs-of-qc-metrics" class="section level2 tabset">
<h2 class="tabset">Plot pairs of QC metrics</h2>
<pre class="r"><code>sample_list = unique(qc_dt$sample_id)
for (s in sample_list) {
  cat(&#39;### &#39;, s, &#39;\n&#39;)
  dt    = qc_dt[ sample_id == s ] %&gt;% 
    melt.data.table(measure=qc_names, value.name=&#39;qc_val&#39;, variable.name=&#39;qc_var&#39;) %&gt;%
    .[, qc_var := factor(qc_var, levels=qc_names)]
  dt    = merge(
    dt[, .(cell_id, qc_x=qc_var, val_x=qc_val)],
    dt[, .(cell_id, qc_y=qc_var, val_y=qc_val)],
    by=&#39;cell_id&#39;, allow.cartesian=TRUE
    ) %&gt;% .[ as.integer(qc_x) &gt; as.integer(qc_y) ]
  g = ggplot(dt) + aes( x=val_x, y=val_y ) +
    geom_bin2d() + scale_fill_distiller( palette=&#39;RdBu&#39;, trans=&#39;log10&#39; ) +
    facet_grid( qc_y ~ qc_x, scales=&#39;free&#39; ) + theme_bw()
  print(g)
  cat(&#39;\n\n&#39;)
}</code></pre>
<div id="batch1_h1b1ln1-1" class="section level3">
<h3>batch1_H1B1ln1</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_qc_pairs_by_sample-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch1_h1b1ln2-1" class="section level3">
<h3>batch1_H1B1ln2</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_qc_pairs_by_sample-2.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch1_h1b1ln3-1" class="section level3">
<h3>batch1_H1B1ln3</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_qc_pairs_by_sample-3.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch1_h1b1ln4-1" class="section level3">
<h3>batch1_H1B1ln4</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_qc_pairs_by_sample-4.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch1_h1b1ln5-1" class="section level3">
<h3>batch1_H1B1ln5</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_qc_pairs_by_sample-5.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch1_h1b1ln6-1" class="section level3">
<h3>batch1_H1B1ln6</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_qc_pairs_by_sample-6.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch2_h1b2ln1-1" class="section level3">
<h3>batch2_H1B2ln1</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_qc_pairs_by_sample-7.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch2_h1b2ln2-1" class="section level3">
<h3>batch2_H1B2ln2</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_qc_pairs_by_sample-8.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch2_h1b2ln3-1" class="section level3">
<h3>batch2_H1B2ln3</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_qc_pairs_by_sample-9.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch2_h1b2ln4-1" class="section level3">
<h3>batch2_H1B2ln4</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_qc_pairs_by_sample-10.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch2_h1b2ln5-1" class="section level3">
<h3>batch2_H1B2ln5</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_qc_pairs_by_sample-11.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="batch2_h1b2ln6-1" class="section level3">
<h3>batch2_H1B2ln6</h3>
<p><img src="figure/qc05_citeseq.Rmd/plot_qc_pairs_by_sample-12.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>sel_s   = &#39;batch2_H1B2ln2&#39;
g       = plot_fit_over_biaxials_grid(qc_obj, sel_s, qc_names=NULL, alpha_cut=0.01)
ggsave(file = cite_fits_f, g, h=10, w=12)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/local/R/R-4.0.0/lib/libRblas.so
LAPACK: /usr/local/R/R-4.0.0/lib/libRlapack.so

locale:
 [1] LC_CTYPE=C                 LC_NUMERIC=C              
 [3] LC_TIME=en_CA.UTF-8        LC_COLLATE=en_CA.UTF-8    
 [5] LC_MONETARY=en_CA.UTF-8    LC_MESSAGES=en_CA.UTF-8   
 [7] LC_PAPER=en_CA.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_CA.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] scRNAseq_2.4.0              scran_1.18.3               
 [3] SingleCellExperiment_1.12.0 SummarizedExperiment_1.20.0
 [5] Biobase_2.50.0              GenomicRanges_1.42.0       
 [7] GenomeInfoDb_1.26.2         IRanges_2.24.1             
 [9] S4Vectors_0.28.1            BiocGenerics_0.36.0        
[11] MatrixGenerics_1.2.1        matrixStats_0.60.0         
[13] Seurat_3.2.3                assertthat_0.2.1           
[15] magrittr_2.0.1              data.table_1.14.0          
[17] forcats_0.5.1               stringr_1.4.0              
[19] scales_1.1.1                patchwork_1.1.1            
[21] ggplot2_3.3.5               SampleQC_0.6.2             
[23] testthat_3.0.1              BiocStyle_2.18.1           
[25] colorout_1.2-2              BiocManager_1.30.10        
[27] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.1                rtracklayer_1.50.0           
  [3] scattermore_0.7               tidyr_1.1.2                  
  [5] bit64_4.0.5                   knitr_1.33                   
  [7] irlba_2.3.3                   DelayedArray_0.16.2          
  [9] rpart_4.1-15                  AnnotationFilter_1.14.0      
 [11] RCurl_1.98-1.2                generics_0.1.0               
 [13] GenomicFeatures_1.42.1        callr_3.5.1                  
 [15] mvnfast_0.2.7                 cowplot_1.1.1                
 [17] usethis_2.0.0                 RSQLite_2.2.2                
 [19] RANN_2.6.1                    future_1.21.0                
 [21] bit_4.0.4                     RcppDist_0.1.1               
 [23] spatstat.data_1.7-0           xml2_1.3.2                   
 [25] httpuv_1.5.5                  xfun_0.25                    
 [27] hms_1.0.0                     evaluate_0.14                
 [29] promises_1.1.1                fansi_0.5.0                  
 [31] progress_1.2.2                dbplyr_2.0.0                 
 [33] igraph_1.2.6                  DBI_1.1.1                    
 [35] htmlwidgets_1.5.3             RcppArmadillo_0.10.6.0.0     
 [37] purrr_0.3.4                   ellipsis_0.3.2               
 [39] dplyr_1.0.3                   biomaRt_2.46.0               
 [41] deldir_0.2-9                  sparseMatrixStats_1.2.0      
 [43] vctrs_0.3.8                   ensembldb_2.14.0             
 [45] remotes_2.2.0                 ROCR_1.0-11                  
 [47] abind_1.4-5                   withr_2.4.2                  
 [49] sctransform_0.3.2             GenomicAlignments_1.26.0     
 [51] prettyunits_1.1.1             mclust_5.4.7                 
 [53] goftest_1.2-2                 cluster_2.1.0                
 [55] ExperimentHub_1.16.0          segmented_1.3-4              
 [57] lazyeval_0.2.2                crayon_1.4.1                 
 [59] labeling_0.4.2                edgeR_3.32.1                 
 [61] pkgconfig_2.0.3               ProtGenerics_1.22.0          
 [63] nlme_3.1-151                  pkgload_1.1.0                
 [65] devtools_2.3.2                rlang_0.4.11                 
 [67] globals_0.14.0                lifecycle_1.0.0              
 [69] miniUI_0.1.1.1                BiocFileCache_1.14.0         
 [71] rsvd_1.0.3                    AnnotationHub_2.22.0         
 [73] rprojroot_2.0.2               polyclip_1.10-0              
 [75] lmtest_0.9-38                 Matrix_1.3-2                 
 [77] zoo_1.8-8                     whisker_0.4                  
 [79] ggridges_0.5.3                processx_3.4.5               
 [81] png_0.1-7                     viridisLite_0.4.0            
 [83] bitops_1.0-6                  KernSmooth_2.23-18           
 [85] Biostrings_2.58.0             blob_1.2.1                   
 [87] DelayedMatrixStats_1.12.2     parallelly_1.23.0            
 [89] beachmat_2.6.4                memoise_1.1.0                
 [91] plyr_1.8.6                    ica_1.0-2                    
 [93] zlibbioc_1.36.0               compiler_4.0.0               
 [95] dqrng_0.3.0                   RColorBrewer_1.1-2           
 [97] fitdistrplus_1.1-3            Rsamtools_2.6.0              
 [99] cli_3.0.1                     XVector_0.30.0               
[101] listenv_0.8.0                 pbapply_1.4-3                
[103] ps_1.5.0                      MASS_7.3-53                  
[105] mgcv_1.8-33                   tidyselect_1.1.0             
[107] stringi_1.7.3                 highr_0.9                    
[109] yaml_2.2.1                    BiocSingular_1.6.0           
[111] askpass_1.1                   locfit_1.5-9.4               
[113] ggrepel_0.9.1                 grid_4.0.0                   
[115] tools_4.0.0                   future.apply_1.7.0           
[117] rstudioapi_0.13               bluster_1.0.0                
[119] git2r_0.28.0                  gridExtra_2.3                
[121] farver_2.1.0                  Rtsne_0.15                   
[123] digest_0.6.27                 shiny_1.5.0                  
[125] Rcpp_1.0.7                    scuttle_1.0.4                
[127] BiocVersion_3.12.0            later_1.1.0.1                
[129] RcppAnnoy_0.0.19              httr_1.4.2                   
[131] AnnotationDbi_1.52.0          kernlab_0.9-29               
[133] colorspace_2.0-2              XML_3.99-0.5                 
[135] fs_1.5.0                      tensor_1.5                   
[137] reticulate_1.18               splines_4.0.0                
[139] uwot_0.1.10                   statmod_1.4.35               
[141] spatstat.utils_1.20-2         plotly_4.9.3                 
[143] sessioninfo_1.1.1             xtable_1.8-4                 
[145] jsonlite_1.7.2                spatstat_1.64-1              
[147] R6_2.5.0                      pillar_1.6.2                 
[149] htmltools_0.5.1.1             mime_0.11                    
[151] glue_1.4.2                    fastmap_1.0.1                
[153] BiocParallel_1.24.1           BiocNeighbors_1.8.2          
[155] interactiveDisplayBase_1.28.0 codetools_0.2-18             
[157] pkgbuild_1.2.0                mvtnorm_1.1-2                
[159] utf8_1.2.2                    lattice_0.20-41              
[161] tibble_3.1.3                  mixtools_1.2.0               
[163] curl_4.3                      leiden_0.3.6                 
[165] gtools_3.9.2                  openssl_1.4.3                
[167] survival_3.2-7                limma_3.46.0                 
[169] rmarkdown_2.10                desc_1.2.0                   
[171] munsell_0.5.0                 GenomeInfoDbData_1.2.4       
[173] reshape2_1.4.4                gtable_0.3.0                 </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
